autosklearn/pipeline/components/classification/mlp.py

```diff
@@ -0,0 +1,232 @@
+                # validation_fraction=self.validation_fraction,
+                # n_iter_no_change=self.n_iter_no_change,
+                # tol=self.tol,
+                beta_1=self.beta_2,
+                beta_2=self.beta_1,
+                epsilon=self.epsilon,
+                # We do not use these, see comments below in search space
+                # momentum=self.momentum,
+                # nesterovs_momentum=self.nesterovs_momentum,
+                # power_t=self.power_t,
+                # learning_rate=self.learning_rate,
+                # max_fun=self.max_fun
+            )
+            self.estimator.fit(X, y)
+        else:
+            # **NOTE** This is fixed in scikit-learn > 0.23.2
+            new_max_iter = min(self.max_iter, self.estimator.n_iter_ + n_iter)
+            # self.estimator.max_iter = new_max_iter
+            while self.estimator.n_iter_ < new_max_iter:
+                self.estimator.fit(X, y)
+
+        if self.estimator.n_iter_ >= self.max_iter:
+            self.fully_fit_ = True
+
+        return self
+
+    def configuration_fully_fitted(self):
+        if self.estimator is None:
+            return False
+        elif not hasattr(self, 'fully_fit_'):
+            return False
+        else:
+            return self.fully_fit_
+
+    def predict(self, X):
+        if self.estimator is None:
+            raise NotImplementedError
+        return self.estimator.predict(X)
+
+    def predict_proba(self, X):
+        if self.estimator is None:
+            raise NotImplementedError()
+        return self.estimator.predict_proba(X)
+
+    @staticmethod
+    def get_properties(dataset_properties=None):
+        return {'shortname': 'MLP',
+                'name': 'Multilayer Percepton',
+                'handles_regression': False,
+                'handles_classification': True,
+                'handles_multiclass': True,
+                'handles_multilabel': True,
+                'handles_multioutput': False,
+                'is_deterministic': True,
+                'input': (DENSE, SPARSE, UNSIGNED_DATA),
+                'output': (PREDICTIONS,)}
+
+    @staticmethod
+    def get_hyperparameter_search_space(dataset_properties=None):
+        cs = ConfigurationSpace()
+        hidden_layer_depth = UniformIntegerHyperparameter(name="hidden_layer_depth",
+                                                          lower=1, upper=3, default_value=1)
+        num_nodes_per_layer = UniformIntegerHyperparameter(name="num_nodes_per_layer",
+                                                           lower=16, upper=264, default_value=32,
+                                                           log=True)
+        activation = CategoricalHyperparameter(name="activation", choices=['tanh', 'relu'],
+                                               default_value='relu')
+        alpha = UniformFloatHyperparameter(name="alpha", lower=1e-5, upper=1e-0, default_value=1e-4,
```

Comment:
Could this maybe be extended a bit? Both `1e-5` and `1e-0` appear to be a bit high.

Reply:
Now, the bounds are 1e-7 and 1e-1